src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -529,8 +529,7 @@ private LocalSymbol MakePatternVariable(TypeSyntax type, VariableDesignationSynt
             }
 
             NamedTypeSymbol container = _scopeBinder.ContainingType;
-            if ((object)container != null && container.IsScriptClass &&
-                (object)_scopeBinder.LookupDeclaredField(designation) != null)
+            if (container is object { IsScriptClass: true } && _scopeBinder.LookupDeclaredField(designation) is object _)
```

Comment:
📝 It seems reasonable to omit the discard variable here.

```suggestion
            if (container is object { IsScriptClass: true } && _scopeBinder.LookupDeclaredField(designation) is object)
```

Reply:
The type `object` does not have a member named `IsScriptClass`, so the type should not be specified for the first.  Agree that the discard can be avoided, but only because you're switching to the is-type expression which has the correct semantics.  You can also write `is { }`.

---
In reply to: [344383593](https://github.com/dotnet/roslyn/pull/39738#discussion_r344383593) [](ancestors = 344383593)

Reply:
@alrz Action item here is to omit the discard during generation. The other issue mentioned is tracked by https://github.com/dotnet/roslyn/pull/39738#discussion_r344381734.

Reply:
as of now we don't know if the type pattern is going to be an immediate child of `is` expression (vs. a subpattern). with the proper type patterns (https://github.com/dotnet/csharplang/issues/2925) we can do this more broadly with a simple langver check.

Reply:
📝 A number of these could be implemented in the Simplifier.